tracer/src/Datadog.Trace/DiagnosticListeners/AspNetCoreDiagnosticObserver.cs

```diff
@@ -503,12 +540,14 @@ private static Span StartMvcCoreSpan(Tracer tracer, Span parentSpan, BeforeActio
                     // If we have a route, overwrite the existing resource name
                     var resourcePathName = SimplifyRoutePattern(
                         routeTemplate,
-                        routeValues,
+                        typedArg.RouteData.Values,
```

Comment:
Had to change this to pass in the `RouteValueDictionary` _directly_ from the `BeforeActionStruct`, as the `actionDescriptor.RouteValues` don't contain _all_ the route values